include/envoy/ssl/connection.h

```diff
@@ -15,39 +15,45 @@ class Connection {
   virtual ~Connection() {}
 
   /**
-   * @return whether the peer certificate is presented.
+   * @return bool whether the peer certificate is presented.
    **/
   virtual bool peerCertificatePresented() const PURE;
 
   /**
-   * @return the URI in the SAN feld of the local certificate. Returns "" if there is no local
-   *         certificate, or no SAN field, or no URI.
+   * @return std::string the URI in the SAN feld of the local certificate. Returns "" if there is no
+   *         local certificate, or no SAN field, or no URI.
    **/
   virtual std::string uriSanLocalCertificate() PURE;
 
   /**
-   * @return the subject field of the local certificate in RFC 2253 format. Returns "" if there is
-   *         no local certificate, or no subject.
+   * @return std::string the subject field of the local certificate in RFC 2253 format. Returns ""
+   *         if there is no local certificate, or no subject.
    **/
   virtual std::string subjectLocalCertificate() const PURE;
 
   /**
-   * @return the SHA256 digest of the peer certificate. Returns "" if there is no peer certificate
-   *         which can happen in TLS (non mTLS) connections.
+   * @return std::string the SHA256 digest of the peer certificate. Returns "" if there is no peer
+   *         certificate which can happen in TLS (non mTLS) connections.
    */
   virtual std::string sha256PeerCertificateDigest() PURE;
 
   /**
-   * @return the subject field of the peer certificate in RFC 2253 format. Returns "" if there is
-   *         no peer certificate, or no subject.
+   * @return std::string the subject field of the peer certificate in RFC 2253 format. Returns "" if
+   *         there is no peer certificate, or no subject.
    **/
   virtual std::string subjectPeerCertificate() const PURE;
 
   /**
-   * @return the URI in the SAN field of the peer certificate. Returns "" if there is no peer
-   *         certificate, or no SAN field, or no URI.
+   * @return std::string the URI in the SAN field of the peer certificate. Returns "" if there is no
+   *         peer certificate, or no SAN field, or no URI.
    **/
   virtual std::string uriSanPeerCertificate() PURE;
+
+  /**
+   * @return std::string the PEM encoded representation of the peer certificate. Returns "" if
+   *         there is no peer certificate or encoding fails.
+   **/
+  virtual std::string peerCertificate() const PURE;
```

Comment:
I have mixed feelings about exposing URL-encoded PEM-encoded peer certificate under this function name (vs raw peer certificate - if we ever need one).

I'm not strongly against it, but wanted to see if anyone else thinks this should be named differently.